plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -81,15 +87,34 @@ fu_flashrom_lspcon_i2c_spi_device_probe (FuDevice *device, GError **error)
 		return FALSE;
 	}
 
-	/* get bus number out of sysfs path */
-	path = fu_udev_device_get_sysfs_path (FU_UDEV_DEVICE (device));
-	regex = g_regex_new (I2C_PATH_REGEX, 0, 0, error);
-	if (regex && g_regex_match_full (regex, path, -1, 0, 0, &info, error)) {
-		self->bus_number = g_ascii_strtoll ( g_match_info_fetch (info, 1),
-					       NULL, 10);
-		return TRUE;
+	/* locate an I2C bus that is a sibling of this device, assumed to be
+	 * the bus the device is connected to */
+	sibling_buses = fu_udev_device_get_siblings_with_subsystem (FU_UDEV_DEVICE (self), "i2c-dev");
+	for (guint i = 0; i < sibling_buses->len; i++) {
+		FuUdevDevice *bus_device = g_ptr_array_index (sibling_buses, i);
+		const gchar *bus_path = fu_udev_device_get_device_file (bus_device);
+
+		if (bus_path == NULL) {
+			g_info ("Bus %s has no device file! Ignoring.",
+			        fu_udev_device_get_sysfs_path (bus_device));
+			continue;
+		}
+
+		self->bus_path = g_strdup (bus_path);
+		if (i < (sibling_buses->len - 1)) {
+			g_info ("Ignoring %u additional sibling buses",
+				sibling_buses->len - i - 1);
+		}
```

Comment:
i'm not really sure what this is doing TBH.

Reply:
Modified slightly to hopefully be clearer. There may be more than one valid sibling bus, but we only care about one so the point is to note that there are additional buses that might have been chosen but weren't.